noir-projects/noir-protocol-circuits/crates/rollup-lib/src/components.nr

```diff
@@ -266,83 +285,140 @@ pub fn compute_tx_effects_hash(
     // The incoming all_public_data_update_requests may have empty update requests in the middle, so we move those to the end of the array.
     let public_data_update_requests =
         get_all_update_requests_for_tx_effects(all_public_data_update_requests);
-    let note_logs_length = combined.note_encrypted_log_preimages_length;
-    let encrypted_logs_length = combined.encrypted_log_preimages_length;
-    let unencrypted_logs_length = combined.unencrypted_log_preimages_length;
-    let note_encrypted_logs_hash = compute_tx_note_logs_hash(combined.note_encrypted_logs_hashes);
-    let encrypted_logs_hash = silo_and_hash_encrypted_logs(combined.encrypted_logs_hashes);
-    let unencrypted_logs_hash = silo_and_hash_unencrypted_logs(combined.unencrypted_logs_hashes);
+
+    let note_encrypted_logs = combined.note_encrypted_logs_hashes.map(|log: LogHash| log.value);
+    let encrypted_logs =
+        combined.encrypted_logs_hashes.map(|log: ScopedLogHash| silo_encrypted_log_hash(log));
+    let unencrypted_logs =
+        combined.unencrypted_logs_hashes.map(|log: ScopedLogHash| silo_unencrypted_log_hash(log));
 
     let mut offset = 0;
+    let mut array_len = 0;
 
     // REVERT CODE
     // upcast to Field to have the same size for the purposes of the hash
-    tx_effects_hash_input[offset] = revert_code as Field;
+    tx_effects_hash_input[offset] = field_from_bytes([REVERT_CODE_PREFIX, 0, revert_code], true);
     offset += 1;
 
     // TX FEE
-    tx_effects_hash_input[offset] = transaction_fee;
+    // TODO(Miranda): how many bytes do we expect tx fee to be? Using 29 for now
```

Comment:
@ TMNT (or any) team: is there a restriction anywhere on the size of the tx fee?